File: web/src/main/java/top/emilejones/hhu/web/service/IRecallService.java
package top.emilejones.hhu.web.service;

import kotlin.Pair;
import top.emilejones.hhu.web.entity.FileNode;
import top.emilejones.hhu.web.entity.TextNode;

import java.util.List;

/**
 * 用来负责处理召回任务
 *
 * @author EmileJones
 */
public interface IRecallService {
    /**
     * 召回和问题相关的文本
     *
     * @param query 问题
     * @return 和问题相关的文本
     */
    List<String> recallText(String query);

    /**
     * 召回和问题相关的节点
     *
     * @param query 问题
     * @return 和问题相关的节点
     */
    List<Pair<FileNode, TextNode>> recallNode(String query);
}

File: infrastructure/textsplitter/structure-extraction/src/main/kotlin/top/emilejones/hhu/textsplitter/service/IRecallService.kt
package top.emilejones.hhu.textsplitter.service

import top.emilejones.hhu.domain.result.FileNode
import top.emilejones.hhu.textsplitter.domain.po.neo4j.Neo4jTextNode

/**
 * 用来负责处理召回任务
 *
 * @author EmileJones
 */
interface IRecallService {

    /**
     * 召回和问题相关的节点
     *
     * @param query 问题
     * @param collectionName milvus的collection名称
     * @param filter 过滤条件
     * @return 和问题相关的节点
     */
    fun recallNode(query: String, collectionName: String, filter: String? = null): List<Neo4jTextNode>

    /**
     * 召回和问题相关的文件节点
     *
     * @param query 问题
     * @param collectionName milvus的collection名称
     * @return 和问题相关的文件节点
     */
    fun recallFileNode(query: String, collectionName: String): List<FileNode>
}
